ethcore/sync/src/chain/propagator.rs

```diff
@@ -45,13 +45,12 @@ fn accepts_service_transaction(client_id: &str) -> bool {
 	// Parity versions starting from this will accept service-transactions
 	const SERVICE_TRANSACTIONS_VERSION: (u32, u32) = (1u32, 6u32);
 	// Parity client string prefix
-	const LEGACY_CLIENT_ID_PREFIX: &'static str = "Parity/v";
-	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity-Ethereum/v";
-
-	let splitted = if client_id.starts_with(LEGACY_CLIENT_ID_PREFIX) {
-		client_id[LEGACY_CLIENT_ID_PREFIX.len()..].split('.')
-	} else if client_id.starts_with(PARITY_CLIENT_ID_PREFIX) {
-		client_id[PARITY_CLIENT_ID_PREFIX.len()..].split('.')
+	const LEGACY_CLIENT_ID_PREFIX: &'static str = "Parity/";
+	const PARITY_CLIENT_ID_PREFIX: &'static str = "Parity-Ethereum/";
+	
+	let idx = client_id.rfind("/v").map(|idx| idx + 1).unwrap_or(client_id.len());
```

Comment:
Actually it should be `idx + 2`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8a040ab9427a3c201706194fcca13d7c
```suggestion
 	let idx = client_id.rfind("/v").map(|idx| idx + 2).unwrap_or(client_id.len());
```

Reply:
Should we change it to idx+"/v" .len() in this case? And change "/v" to a constant?

Reply:
Yes, makes sense.